rpcserver.go

```diff
@@ -4172,7 +4139,7 @@ func (r *rpcServer) DeleteAllPayments(ctx context.Context,
 
 	rpcsLog.Debugf("[DeleteAllPayments]")
 
-	if err := r.server.chanDB.DeleteAllPayments(); err != nil {
+	if err := r.server.chanDB.DeletePayments(); err != nil {
```

Comment:
previously this wouldn't affect the control tower state, but now it appears it will?

Reply:
That's true, good catch! 

Should we make it delete completed payments, but leave in-flight ones?

Reply:
Fixed: https://github.com/lightningnetwork/lnd/pull/2761/commits/b6b24debd400668f0a8f0a19d2a7766eec48f362

Reply:
Another small fixup (https://github.com/lightningnetwork/lnd/pull/2761/commits/6c0c6b89aa1027c117416b1f9c6f7e658e40af46) and test (https://github.com/lightningnetwork/lnd/pull/2761/commits/8d5c185f0d1808f4f30e5a03c727d055b6e50a25)